test/components/retrievers/test_auto_merging_retriever.py

```diff
@@ -0,0 +1,115 @@
+from unittest import mock
+
+import pytest
+
+from haystack import Document
+from haystack_experimental.components.splitters import HierarchicalDocumentSplitter
+from haystack_experimental.components.retrievers.auto_merging_retriever import AutoMergingRetriever
+from haystack.document_stores.in_memory import InMemoryDocumentStore
+
+
+class TestAutoMergingRetriever:
+    def test_init_default(self):
+        retriever = AutoMergingRetriever(InMemoryDocumentStore())
+        assert retriever.threshold == 0.5
+
+    def test_init_with_parameters(self):
+        retriever = AutoMergingRetriever(InMemoryDocumentStore(), threshold=0.7)
+        assert retriever.threshold == 0.7
+
+    def test_init_with_invalid_threshold(self):
+        with pytest.raises(ValueError):
+            AutoMergingRetriever(InMemoryDocumentStore(), threshold=-2)
+
+    def test_to_dict(self):
```

Comment:
Let's always add a test does a full pipeline serde roundtrip, i.e., serializes a pipeline with the component and deserializes it from the dumped string. Same for the splitter.

Reply:
good point - I forgot that test